js/components/grid/course_panel.js.jsx

```diff
@@ -26,42 +26,82 @@ class Course extends React.Component {
     super(props);
     this.toggleSelect = this.toggleSelect.bind(this);
     this.removeCourse = this.removeCourse.bind(this);
+    this.parseLectures = this.parseLectures.bind(this);
     this.state = {
-      selected: false
+      selected: false,
+      courseInfo: {}
     }
   }
 
   componentDidMount() {
-    //.ajax.data option add additional data to the requestcan be given as an object with parameters and values to submit,
-    // jQuery's ajax function just appends data to the URL as URL parameters for GET requests
-    // Below the deprecated method of retrieving the course information
-    // var course;
-    // $.ajax({
-    //     url: 'course',
-    //     dataType: 'json',
-    //     data: {name : courseCode},
-    //     async: false,
-    //     success: function (data) {
-    //         course = data;
-    //     },
-    //     error: function () {
-    //         throw 'No course file';
-    //     }
-    // });
-    // console.log(course);
-
     fetch(
       'course?name=' + this.props.courseCode, // url to which the AJAX request is sent to
     )
     .then(response => response.json()) //datatype
     .then(data => {
-        console.log(data)
-      // Parse the information
-      // Check data.fall/spring/yearSession
-      // loop through data.fall/spring/yearSession.lectures: check for section(the section code) to remove duplicates
-      // check times: timeField[0] is day index, timeField[1] is hour (there's an extra one for every half hour)
-      // {courseCode: , F: [lectureObject, lectureObject, lectureObjext], S:[], Y:[] }
+        let course = {courseCode: "", F: [], S:[], Y:[]}
+        course.courseCode = data.name;
+        course.F = course.F.concat(this.parseLectures(data.fallSession.lectures),
+                                    this.parseLectures(data.fallSession.tutorials));
+        course.S = course.S.concat(this.parseLectures(data.springSession.lectures),
+                                    this.parseLectures(data.springSession.tutorials));
+        course.Y = course.Y.concat(this.parseLectures(data.yearSession.lectures),
+                                    this.parseLectures(data.yearSession.tutorials));
+        this.setState({courseInfo: course});
+    });
+  }
+
+  parseLectures(lectures) {
+    // Remove duplicated lecture sections
+    let allLectures = lectures.filter((lecture, index, lectures) => {
+      return (lectures.map(lect => lect.section).indexOf(lecture.section) == index)
     });
+    let parsedLectures = [];
+
+    let days = {0: 'M', 1: 'T', 2: 'W', 3: 'R', 4: 'F'};
```

Comment:
Should days also be initalized like this? 
`let days = {
  0: 'M',
  1: 'T',
  2: 'W',
  3" 'R',
  4: 'F'
};
`
Or is the multi-line initialization unnecessary because the object is already easily readable?